src/sentry/testutils/hybrid_cloud.py

```diff
@@ -143,6 +137,20 @@ def assert_org_member_mapping_not_exists(self, org_member: OrganizationMember):
 class SimulatedTransactionWatermarks(threading.local):
     state: dict[str, int] = {}
 
+    @staticmethod
+    def get_transaction_depth(connection: BaseDatabaseWrapper) -> int:
+        total = len(connection.savepoint_ids)
+        if connection.in_atomic_block:
+            total += 1
```

Comment:
Will there ever be `savepoint_ids` for a connection without a transaction block?

Reply:
Nope!